packages/flutter_tools/lib/src/reporting/pii_regexp.dart

```diff
@@ -0,0 +1,77 @@
+/// framework do not contain PII.   The regular expressions here are
+/// intentionally broad to make sure they get everything, possibly at the
+/// expense of readability for the error message.  Use care when reducing
+/// their scope.
+library;
+
+/// Contain an ordered list of regular expressions to be applied in sequence
+/// with [applyTo] to a string, with any matches being substituted with
+/// [substitution].  While a single regular expression could be
+/// used in place of this class, this is intending to make the regular
+/// expressions being applied more comprehensible to non-expert regular
+/// expression authors and to make it very simple to add or remove individual
+/// cases.
+class _RegExpList {
+  _RegExpList(List<String> uncompiledRegexps, this.substitution)
+      : _regExps = uncompiledRegexps.map((String s) => RegExp(s)).toList();
+
+  /// This list should be usually be ordered from more specific to more general
+  /// cases in the event the regular expressions overlap.
+  final List<RegExp> _regExps;
+  final String substitution;
+
+  String applyTo(String input) => _regExps.fold(
+      input, (String previousInput, RegExp r) => previousInput.replaceAll(r, substitution));
+}
+
+/// An ordered list of regular expressions to be substituted out and replaced
+/// with a generic '<path>' string. To be applied in sequence.
+/// The regular expressions here intentionally eat all characters after a
+/// path is identified, with the hope of catching all potential complex
+/// pathnames and assuming that most error messages will have any paths last.
+/// For example: `throw Exception('error in: $fileName')`
+final _RegExpList _piiPathRegexps = _RegExpList(<String>[
+  // C:\Some\Things\Including Spaces\Too
+  r'\w:\\.*',
+  // \\Windows\Shares\Look\Like This
+  r'\\\\\S+\\.*',
+  // ..\StartsWithDots\Windows Path
+  r'[.]+\\+\S*\\.*',
+  // A\Relative\Windows Path
+  r'\b\S+\\.*',
+  // ../StartsWithDots/Unix path
+  r'[.]+/\S*/.*',
+  // /AUnix/path
+  r'/(\S|[.])+/.*',
+  // a/unix\ relative path/possibly-containing-strange-characters
+  r'\b(\S|[.])+/.*',
+], '<path>');
+
+/// An ordered list of regular expressions to be substituted out and replaced
+/// with a generic '<filename>' string.  To be applied after
+/// all paths from [_piiPathRegexps] are applied.   This is somewhat less
+/// all-encompassing than the path finder, above, just to try and grab things
+/// that could be file names but without being quite so greedy or depending
+/// on the presence of path separators.
+final _RegExpList _piiFileRegexps = _RegExpList(<String>[
+  // foo.dart
+  // something.exe
+  // code.js
+  r'\b\S+[.]\w{1,4}\b',
+], '<filename>');
+
+/// Attempt to remove things that look like they could be PII from the
+/// given [message].
+/// Returns the filtered string.
+/// Not suitable for pre-scrubbed strings that intentionally include things
+/// that look like filenames.
+String filterPiiFromErrorMessage(String message) => <_RegExpList>[
```

Comment:
This will at least need an allowlist for files and paths within the tool's artifact cache so that we can discern when crashes are triggered by errors in downloading, extracting, or reading those files. Paths relative to the tool's artifact cache do not contain PII.

Reply:
What is the artifact cache?  `${FLUTTER_ROOT}/bin/cache`?

Reply:
Yes, though @christopherfujino might have more locations in mind that shouldn't be sanitized. It's arguable that this could be widened to anything under `${FLUTTER_ROOT}`.

Reply:
agreed, I would say anything under FLUTTER_ROOT is part of the Flutter SDK and we shouldn't redact, at least not relative to the FLUTTER_ROOT.